src/Providers/Stripe/Card/Service.php

```diff
@@ -0,0 +1,185 @@
+    public function remove(Token $cardToken, $customerToken = null): void
+    {
+        switch ($cardToken->type) {
+            case StripeToken::PAYMENT_METHOD:
+                /** @var PaymentMethod $paymentMethod */
+                $paymentMethod = PaymentMethod::retrieve($cardToken->value);
+
+                $paymentMethod->detach();
+
+                break;
+            case StripeToken::CARD:
+                if ($customerToken->type !== StripeToken::CUSTOMER) {
+                    throw new \Exception('You need to provide a valid customer Token.');
+                }
+
+                StripeCustomer::deleteSource($customerToken->value, $cardToken->value);
+
+                break;
+            default:
+                throw new \Exception('You need to provide a valid card/payment method Token.');
+        }
+    }
+
+    /**
+     * Tries create a card using the Stripe SDK.
+     *
+     * @param Token $token
+     * @param array $params
+     *
+     * @return Card
+     *
+     * @throws Exception
+     * @throws CardException
+     */
+    public function create(Token $token, array $params = []): Card
+    {
+        switch ($token->type) {
+            case StripeToken::UNDEFINED:
+                $params = array_merge(
+                    [
+                        'usage' => 'on_session',
+                        'payment_method_types' => ['card'],
+                    ],
+                    $params
+                );
+
+                $response = SetupIntent::create($params);
+
+                break;
+            case StripeToken::SETUP_INTENT:
+                $response = SetupIntent::retrieve($token->value);
+
+                break;
+            default:
+                throw new \Exception('You either need to pass a null Token or a setup intent one.');
+        }
+
+        if ($response->status !== SetupIntent::STATUS_SUCCEEDED) {
+            $token = new StripeToken($response->client_secret);
+
+            throw new CardException($token);
+        }
+
+        /** @var PaymentMethod $paymentMethod */
+        $paymentMethod = PaymentMethod::retrieve($response->payment_method);
+
+        return new Card(
+            $paymentMethod->id,
+            $paymentMethod->card->last4,
+            $paymentMethod->card->brand,
+            $paymentMethod->card->funding,
+            $paymentMethod->card->exp_month,
+            $paymentMethod->card->exp_year
+        );
+    }
+
+    /**
+     * Tries get cards using the Stripe SDK.
+     *
+     * @param Token $token
+     *
+     * @return array[Card]
+     */
+    public function index(Token $token): array
+    {
+        $cards = [];
+
+        switch ($token->type)
+        {
+            case StripeToken::CUSTOMER:
+                $paymentMethods = PaymentMethod::all([
+                    "customer" => $token->value,
+                    "type" => "card",
+                ]);
+
+                if (isset($paymentMethods->data) && $paymentMethods->data) {
```

Comment:
never like to see issets

Reply:
I took the code directly from what existed on the API3 but I'm happy to modify it if you want me to :-)